rpcs3/Emu/system_utils.cpp

```diff
@@ -154,6 +156,94 @@ namespace rpcs3::utils
 		return rap_path;
 	}
 
+	std::string get_c00_unlock_edat_path(const std::string& title_id, const std::string& content_id)
+	{
+		const std::string home_dir = get_hdd0_dir() + "home";
+
+		std::string edat_path;
+
+		for (auto&& entry : fs::dir(home_dir))
+		{
+			if (entry.is_directory && check_user(entry.name))
+			{
+				edat_path = fmt::format("%s/%s/exdata/%s/%s.edat", home_dir, entry.name, title_id, content_id);
+				if (fs::is_file(edat_path))
+				{
+					return edat_path;
+				}
+			}
+		}
+
+		// Return a sample path tested for logging purposes
+		return edat_path;
+	}
+
+	bool verify_c00_unlock_edat(const std::string& title_id, const std::string& content_id)
+	{
+		const std::string edat_path = rpcs3::utils::get_c00_unlock_edat_path(title_id, content_id);
+		std::string rap_path        = rpcs3::utils::get_rap_file_path(content_id);
+		u128 k_licensee             = get_default_self_klic();
```

Comment:
EDAT are opened with current KLIC, see lv2_file::open_raw().

Reply:
Impossible, PS3 doesn't pass klic in sceNpDrmVerifyUpgradeLicense and it doesn't call sceNpDrmIsAvailable for this EDAT either. I've tested three different C00 games and they work so I believe this is the intended behavior.

Reply:
That's why I suggested you to look at lv2_file::open_raw() implementation, to see how it's done.

Reply:
I did check earlier and it calls `g_fxo->get<loaded_npdrm_keys>();` which is filled by `sceNpDrmIsAvailable`. I don't see what you're getting at.

Reply:
Hmm ok, then should be fine.